### DISCUSSION TITLE
Is there any way to hide protected selectors from the selectorManager?

### DISCUSSION THREAD

**Discussion by glaprida** [GrapesJS/grapesjs#5500] [Q&A]
I know private selectors don't show; but also you can't change any values to them, and `componentFirst: false` don't play well with private selectors too.

I am trying to get the best of both worlds; having some class pre-applied to some components (but don't show it on the selectorManager) while also being able to modify it's style values with `componentFist: false` so that it automatically updates all components with that class.

Is there a way to hide protected selectors without writing a whole custom selectorManager from scratch?

**Comment by mosh-tudor** [comment]
> Is there a way to hide protected selectors without writing a whole custom selectorManager from scratch?

You can try using data attributes (yeah, a bit hacky, but at least it works).

**Reply by glaprida** [reply]
how would you do that?

**Reply by mosh-tudor** [reply]
> some class pre-applied to some components (but don't show it on the selectorManager)

JS:
```js
  blockManager.add('grid', {
    label: `
      <div class='gjs-block-icon'>${gridIcon}</div>
      <div>Grid</div>
    `,
    category: "Layout",
    content: '<div class="my-grid-component" data-random-name="grid">..//..</div>',
  });
```

CSS:
```css
[data-random-name="grid"] {
  border: 1px solid red;
}
```

**Reply by glaprida** [reply]
That won't work for me... I need the class to be hidden, pre-applied, but also to be able to modify it through the styles manager (with componentFirst: false)

As far as I understand I can't modify the values of a [data-attribute] selector

**Comment by artf** [comment]
> Is there a way to hide protected selectors without writing a whole custom selectorManager from scratch?

Unfortunately, this would be the only recommended way right now, without incurring hacky solutions (eg. update selector tags DOM on target change).
In the next release, I'll add a new class name for protected selector elements.

**Reply by glaprida** [reply]
That would be sooo great ! Thanks!
